godtools/App/Features/Dashboard/Presentation/Tools/ToolsViewModel.swift

```diff
@@ -67,17 +66,17 @@ class ToolsViewModel: ObservableObject {
         
         getCurrentAppLanguageUseCase
             .getLanguagePublisher()
+            .receive(on: DispatchQueue.main)
             .assign(to: &$appLanguage)
         
-        Publishers.CombineLatest4(
-            $didPullToRefresh.eraseToAnyPublisher(),
-            $appLanguage.eraseToAnyPublisher(),
-            $toolFilterCategorySelection.eraseToAnyPublisher(),
-            $toolFilterLanguageSelection.eraseToAnyPublisher()
+        Publishers.CombineLatest3(
+            $appLanguage.eraseToAnyPublisher().dropFirst(),
+            $toolFilterCategorySelection.eraseToAnyPublisher().dropFirst(),
+            $toolFilterLanguageSelection.eraseToAnyPublisher().dropFirst()
         )
-        .flatMap({ (pullToRefresh, appLanguage, toolFilterCategory, toolFilterLanguage) -> AnyPublisher<(ViewToolsDomainModel, [SpotlightToolListItemDomainModel]), Never> in
+        .map{ (appLanguage, toolFilterCategory, toolFilterLanguage) -> AnyPublisher<(ViewToolsDomainModel, [SpotlightToolListItemDomainModel]), Never> in
                                     
-            return Publishers.CombineLatest(
+            Publishers.CombineLatest(
```

Comment:
Hey @rachaelblue from my understanding flatMap will create a new Publisher from an Upstream publisher where as map won't create a new Publisher but instead change the values produced in the Upstream publisher and produce Upstream.Output.  

As to why switchToLatest() isn't available on flatMap is not clearly understood to me.  There are some subtle differences in the return types for flatMap and map where flatMap produces the generic NewPublisher and map produces the generic Upstream.Output.

I also noticed in the documentation when initializing switchToLatest it would take a type Upstream.

There's a lot of generics happening and I need to spend more time investigating the differences between map, flatMap, etc.